Syndication/AzureStack.MarketplaceSyndication.psm1

```diff
@@ -151,10 +158,39 @@ function Get-Dependency {
 
     Write-Host "`nDownloading product: $productid" -ForegroundColor DarkCyan
     if ($PSBoundParameters.ContainsKey('azCopyDownloadThreads')) {
-        Download-Product -productid $productid -resourceGroup $resourceGroup -azureEnvironment $azureEnvironment -azureSubscriptionID $azureSubscriptionID -registration $registration -token $token -destination $destination -azCopyDownloadThreads $azCopyDownloadThreads
+        Download-Product -productid $productid -resourceGroup $resourceGroup -azureEnvironment $azureEnvironment -azureSubscriptionID $azureSubscriptionID -registration $registration -token $token -destination $destination -azCopyDownloadThreads $azCopyDownloadThreads -azCopyPath $azCopyPath
     } else {
-        Download-Product -productid $productid -resourceGroup $resourceGroup -azureEnvironment $azureEnvironment -azureSubscriptionID $azureSubscriptionID -registration $registration -token $token -destination $destination
+        Download-Product -productid $productid -resourceGroup $resourceGroup -azureEnvironment $azureEnvironment -azureSubscriptionID $azureSubscriptionID -registration $registration -token $token -destination $destination -azCopyPath $azCopyPath
+    }
+}
+
+function ValidateOrGetAzCopyPath($azCopyPath) {
+    # getting azcopy path from environment variables user didn't provide azCopyPath parameter 
+    if ([string]::IsNullOrEmpty($azCopyPath)){ 
+        $azCopyPath = (Get-Command 'azcopy' -ErrorAction Ignore).Source
+        if ([string]::IsNullOrEmpty($azCopyPath)){
+            Write-Verbose "Please download AzCopy V10 and add its path to Environment variables path or pass AzCopy V10 path as an addtional parameter (-azCopyPath), canceling" -verbose
+            return $null
+        }
+    }else{
+        # if user just provided the directory containing azcopy.exe
+        if (-not ($azCopyPath -match ".exe$")){ $azCopyPath = $azCopyPath.TrimEnd("\")+"\azcopy.exe" }
+    }
+    try{
+        # invoking azcopy command to validate if it exists and getting info such as version
+        $azCopyInfo = & $azCopyPath
+        # getting version data from other details
+        $azCopyVersion = $azCopyInfo | Where-Object {$_ -match "AzCopy [0-9]+\.[0-9]+\.[0-9]+"} | Foreach {$matches[0]}
```

Comment:
better to check with version [System.Version]

Reply:
--version command doesn't work with all versions of AzCopy